cmd/bosun/database/state_data.go

```diff
@@ -447,3 +516,58 @@ func (d *dataAccess) transact(conn redis.Conn, f func() error) error {
 	}
 	return nil
 }
+
+// CleanupCleanupOldRenderedTemplates will in a loop purge any old rendered templates
+func (d *dataAccess) CleanupOldRenderedTemplates(olderThan time.Duration) {
+	// run after 5 minutes (to let bosun stabilize)
+	// and then every hour
+	time.Sleep(time.Minute * 5)
+	for {
+		conn := d.Get()
+		slog.Infof("Cleaning out old rendered templates")
+		earliestOk := time.Now().UTC().Add(-1 * olderThan)
+		func() {
+			toPurge := []int64{}
+			keys, err := d.GetRenderedTemplateKeys()
+			if err != nil {
+				slog.Error(err)
+				return
+			}
+			for _, key := range keys {
+				parts := strings.Split(key, ":")
+				if len(parts) != 2 {
+					slog.Errorf("Invalid rendered template redis key found: %s", key)
+					continue
+				}
+				id, err := strconv.ParseInt(parts[1], 10, 64)
+				if err != nil {
+					slog.Error(err)
+					continue
+				}
+				state, err := d.getIncident(id, conn)
+				if err != nil {
+					if strings.Contains(err.Error(), "nil returned") {
```

Comment:
strings.Contains on error, combined with the match not being a constant doesn't seem safe in the long run.

Reply:
Yeah, the problem there is that the error will already be wrapped at that point, and the inner error is not accessible. I guess I can also make a package function for that test though.